Jenkinsfile

```diff
@@ -0,0 +1,324 @@
+            pip install ../cryptography
+            pip install pyopenssl service_identity pycrypto
+            pip install -e .
+            python -m twisted.trial src/twisted
+        """
+    ],
+    [
+        downstreamName: 'paramiko',
+        label: 'docker',
+        imageName: 'pyca/cryptography-runner-ubuntu-rolling',
+        script: """#!/bin/bash -xe
+            git clone --depth=1 https://github.com/paramiko/paramiko.git paramiko
+            cd paramiko
+            virtualenv .venv
+            source .venv/bin/activate
+            pip install ../cryptography
+            pip install -e .
+            pip install -r dev-requirements.txt
+            inv test
+        """
+    ],
+]
+
+def checkout_git(label) {
+    def script = ""
+    if (env.BRANCH_NAME.startsWith('PR-')) {
+        script = """
+        git clone --depth=1 https://github.com/pyca/cryptography.git cryptography
+        cd cryptography
+        git fetch origin +refs/pull/${env.CHANGE_ID}/merge:
+        git checkout -qf FETCH_HEAD
+        git rev-parse HEAD
+        """
+    } else {
+        script = """
+        git clone --depth=1 https://github.com/pyca/cryptography.git cryptography
+        cd cryptography
+        git checkout ${env.BRANCH_NAME}
+        git rev-parse HEAD
+        """
+    }
+    if (label.contains("windows")) {
+        bat script
+    } else {
+        sh script
+    }
+}
+def build(toxenv, label, imageName) {
+
+    try {
+        timeout(time: 30, unit: 'MINUTES') {
+
+            checkout_git(label)
+
+            withCredentials([string(credentialsId: 'cryptography-codecov-token', variable: 'CODECOV_TOKEN')]) {
+                withEnv(["LABEL=$label", "TOXENV=$toxenv", "IMAGE_NAME=$imageName"]) {
+                    if (label.contains("windows")) {
+                        bat """
+                            cd cryptography
+                            @set PATH="C:\\Python27";"C:\\Python27\\Scripts";%PATH%
+                            @set CRYPTOGRAPHY_WINDOWS_LINK_OPENSSL110=1
+                            if %TOXENV% == py26 (
+                                @set PYTHON="C:\\Python26\\python.exe"
+                            )
+                            if %TOXENV% == py27 (
+                                @set PYTHON="C:\\Python27\\python.exe"
+                            )
+                            if %TOXENV% == py33 (
+                                @set PYTHON="C:\\Python33\\python.exe"
+                            )
+                            if %TOXENV% == py34 (
+                                @set PYTHON="C:\\Python34\\python.exe"
+                            )
+                            if %TOXENV% == py35 (
+                                @set PYTHON="C:\\Python35\\python.exe"
+                            )
+                            if %TOXENV% == py36 (
+                                @set PYTHON="C:\\Python36\\python.exe"
+                            )
```

Comment:
This question applies to all the `if` statements in here, doing the `ifs` in groovy seems way better.